### PR TITLE
Add hypre calls for init and finalize

### PR DESCRIPTION
**Pull request by lebuller**
Add calls to `HYPRE_Init` and `HYPRE_Finalize` in the `HypreStructuredSolver` which is necessary for the current hypre master (post release 2.28) branch and should be done for hypre standard practice.

### CODE REVIEW

File: cajita/src/Cajita_HypreStructuredSolver.hpp [hunk 1]
```diff
@@ -120,6 +120,8 @@ class HypreStructuredSolver
         : _comm( layout.localGrid()->globalGrid().comm() )
         , _is_preconditioner( is_preconditioner )
     {
+        HYPRE_Init();
```
Comment (dalg24) [inline cajita/src/Cajita_HypreStructuredSolver.hpp:123]: Is it legal to call `HYPRE_Init()` multiple times in a program?

Comment (lebuller) [reply cajita/src/Cajita_HypreStructuredSolver.hpp:123]: It is legal to call `HYPRE_Init()` and then `HYPRE_Finalize()` followed by another call to `HYPRE_Init()` to create multiple different HYPRE solvers at discrete times. I am not sure if it is legal or if there is an issue with multiple calls to `HYPRE_Init()` before a call to `HYPRE_Finalize()`. I do not see any specific info on such a case in the documentation for hypre.

**Comment by streeve** [comment]
From the call: add the init/final to each example rather than in the solver, with clear comments on user responsibility for that

**Comment by streeve** [comment]
Looks good, but needs to be formatted

**Comment by streeve** [comment]
@lebuller you can format with `make cabana-format`

File: cajita/unit_test/tstHypreStructuredSolver2d.hpp [hunk 2]
```diff
@@ -196,6 +199,9 @@ poissonTest( const std::string& solver_type, const std::string& precond_type,
         for ( int j = owned_space.min( Dim::J ); j < owned_space.max( Dim::J );
               ++j )
             EXPECT_FLOAT_EQ( lhs_host( i, j, 0 ), lhs_ref_host( i, j, 0 ) );
+
+    // finalize hypre
+    HYPRE_Finalize();
```
Comment (dalg24) [inline cajita/unit_test/tstHypreStructuredSolver2d.hpp:204]: Does the solver need to be out of scope before HYPRE is finalized?

File: cajita/unit_test/tstHypreStructuredSolver2d.hpp [hunk 3]
```diff
@@ -203,52 +203,72 @@ poissonTest( const std::string& solver_type, const std::string& precond_type,
 //---------------------------------------------------------------------------//
 TEST( structured_solver, pcg_none_test )
 {
+    HYPRE_Init();
     poissonTest( "PCG", "none", TEST_MEMSPACE{} );
+    HYPRE_Finalize();
```
Comment (dalg24) [inline cajita/unit_test/tstHypreStructuredSolver2d.hpp:208]: Did you consider doing it in `main()` instead?

Comment (streeve) [reply cajita/unit_test/tstHypreStructuredSolver2d.hpp:208]: I agree - a single new scope within the poisson function would be cleaner

File: cajita/unit_test/tstHypreStructuredSolver2d.hpp (hunk 2 above)
Comment (streeve) [reply cajita/unit_test/tstHypreStructuredSolver2d.hpp]: @lebuller can you point to the docs where they discuss this kind of thing?

**Review by streeve — APPROVED** [review]

**Comment by streeve** [comment]
format this please

**Comment by streeve** [comment]
HIP timeout unrelated

**Pull request merged**